File: doc-examples/example-java/src/main/java/example/BlobServiceFactory.java
package example;

import com.azure.core.credential.TokenCredential;
import com.azure.storage.blob.BlobServiceClient;
import com.azure.storage.blob.BlobServiceClientBuilder;
import io.micronaut.context.annotation.Factory;
import jakarta.inject.Singleton;
import org.jspecify.annotations.NonNull;

//tag::class[]
@Factory
public class BlobServiceFactory {  // <1>

    @Singleton
    public BlobServiceClient blobServiceClient(@NonNull TokenCredential tokenCredential) {  // <2>
        return new BlobServiceClientBuilder()  // <3>
                .credential(tokenCredential)
                .endpoint(System.getenv("AZURE_BLOB_ENDPOINT"))
                .buildClient();
    }
}
//end::class[]

File: doc-examples/example-kotlin/src/main/kotlin/example/BlobServiceFactory.kt
package example

import com.azure.core.credential.TokenCredential
import com.azure.storage.blob.BlobServiceClient
import com.azure.storage.blob.BlobServiceClientBuilder
import io.micronaut.context.annotation.Factory
import jakarta.inject.Singleton

//tag::class[]
@Factory
class BlobServiceFactory { // <1>

    @Singleton
    fun blobServiceClient(tokenCredential: TokenCredential): BlobServiceClient { // <2>
        return BlobServiceClientBuilder() // <3>
            .credential(tokenCredential)
            .endpoint(System.getenv("AZURE_BLOB_ENDPOINT"))
            .buildClient()
    }
}
//end::class[]
